### PR TITLE
DYN-6602 Make Dynamo TSpline nodes OOTB and hide experiment toggle under feature flag

### PR DESCRIPTION
**Pull request by QilongTang**
### Purpose

DYN-6602 Make Dynamo TSpline nodes OOTB and hide experiment toggle under feature flag.
image

Tested cases covering:

- Feature flag off - TSpline nodes available OOTB
- Feature flag on - Toggle visible and use able to control show/hide TSpline nodes based on toggle value

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Make Dynamo TSpline nodes OOTB and hide experiment toggle under feature flag

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -728,6 +729,7 @@ protected DynamoModel(IStartConfiguration config)
                     }
                     catch (Exception e) { Logger.LogError($"could not start feature flags manager {e}"); };
                 });
+                DynamoFeatureFlagsManager.FlagsRetrieved += HandleFeatureFlags;
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs:732]: Add feature flag event based TSpline nodes show/hide

**Review by zeusongit — APPROVED** [review]

**Comment by Amoursol** [comment]
@QilongTang did we want to hide the entire Experimental section also if we no longer have anything inside it? i.e. - it shows if we have something, is hidden if not?

**Review by reddyashish — APPROVED** [review]

**Comment by mjkkirschner** [comment]
> @QilongTang did we want to hide the entire Experimental section also if we no longer have anything inside it? i.e. - it shows if we have something, is hidden if not?

@sm6srw is going to add paneling nodes to it.

File: src/DynamoCore/Configuration/PreferenceSettings.cs [hunk 2]
```diff
@@ -1152,6 +1155,21 @@ internal void InitializeNamespacesToExcludeFromLibrary()
             }
         }
 
+        /// <summary>
+        /// Update namespaces to exclude from Library based on feature flags
+        /// </summary>
+        internal void UpdateNamespacesToExcludeFromLibrary()
+        {
+            // When the experiment toggle is disabled by feature flag, include the TSpline namespace from the library OOTB.
+            if (!DynamoModel.FeatureFlags?.CheckFeatureFlag("IsTSplineNodesExperimentToggleVisible", false) ?? false)
```
Comment (aparajit-pratap) [inline src/DynamoCore/Configuration/PreferenceSettings.cs:1164]: Why not use `preferenceSettings.IsTSplineNodesExperimentToggleVisible` property here?

**Comment by aparajit-pratap** [comment]
> > > @QilongTang did we want to hide the entire Experimental section also if we no longer have anything inside it? i.e. - it shows if we have something, is hidden if not?
> > 
> > 
> > @sm6srw is going to add paneling nodes to it.
> 
> Agreed, I think we will have more experimental features in the future, e.g. new App home

As far as I understand, this PR isn't doing it the way @Amoursol is suggesting. The display of the experimental expander purely depends on whether the feature flag is on or off, not on whether there are experimental features or not. I think the experimental section should be displayed based on whether `NamespacesToExcludeFromLibrary` is empty or not.

**Pull request merged**